src/api/uploads/MultiputPart.js

```diff
@@ -26,38 +30,289 @@ class MultiputPart extends Base {
         | typeof PART_STATE_UPLOADED;
     timing: Object;
     uploadedBytes: number;
+    uploadUrl: string;
+    onProgress: Function;
+    onSuccess: Function;
+    onError: Function;
+    data: Object;
+    config: MultiputConfig;
+    id: number;
+    retryTimeout: ?number;
+    blob: ?Blob;
+    rangeEnd: number;
+    startTimestamp: number;
+    getPartsState: Function;
+    getNumPartsUploading: Function;
 
     /**
      * [constructor]
      *
-     * @param {Object} options
+     * @param {Options} options
      * @param {number} index - 0-based index of this part in array of all parts
      * @param {number} offset - Starting byte offset of this part's range
      * @param {number} size - Size of this part in bytes
+     * @param {number} sessionId
+     * @param {Object} sessionEndpoints
+     * @param {MultiputConfig} config
+     * @param {Function} getPartsState
+     * @param {Function} getNumPartsUploading
+     * @param {Function} [onSuccess]
+     * @param {Function} [onProgress]
+     * @param {Function} [onError]
      * @return {void}
      */
-    constructor(options: Object, index: number, offset: number, size: number): void {
-        super(options);
+    constructor(
+        options: Options,
+        index: number,
+        offset: number,
+        size: number,
+        sessionId: string,
+        sessionEndpoints: Object,
+        config: MultiputConfig,
+        getPartsState: Function,
+        getNumPartsUploading: Function,
+        onSuccess?: Function,
+        onProgress?: Function,
+        onError?: Function
+    ): void {
+        super(options, sessionEndpoints, config);
 
         this.index = index;
         this.numDigestRetriesPerformed = 0;
         this.numUploadRetriesPerformed = 0;
         this.offset = offset;
-        this.sha256 = null;
         this.size = size;
         this.state = PART_STATE_NOT_STARTED;
         this.timing = {};
         this.uploadedBytes = 0;
+        this.data = {};
+        this.uploadUrl = `${this.uploadHost}/api/2.0/files/upload_sessions/${sessionId}`;
```

Comment:
The uploadUrl should come from sessionEndpoints

Reply:
done